docs/faq.md

```diff
@@ -310,3 +310,93 @@ The currently available K/V cache quantization types are:
+
+Memory calculations depend on the architecture of the model and sometimes it's not correct.  This can lead to the runner terminating from an OOM.  This may be mitigated by one or more of the following:
+
+1. Set [`OLLAMA_GPU_OVERHEAD`](https://github.com/ollama/ollama/blob/5f8051180e3b9aeafc153f6b5056e7358a939c88/envconfig/config.go#L237) to give llama.cpp a buffer to grow in to (eg, `OLLAMA_GPU_OVERHEAD=536870912` to reserve 512M).  The exact value depends on model/GPU.
+2. Enable flash attention by setting [`OLLAMA_FLASH_ATTENTION=1`](https://github.com/ollama/ollama/blob/5f8051180e3b9aeafc153f6b5056e7358a939c88/envconfig/config.go#L236) in the server environment.  Flash attention is a more efficient use of memory and may reduce memory pressure.  See above.
+3. Reduce the number layers that ollama thinks it can offload to the GPU, see [here](https://github.com/ollama/ollama/issues/6950#issuecomment-2373663650).
+4. In Linux, set `GGML_CUDA_ENABLE_UNIFIED_MEMORY=1`.  This will allow the GPU to offload to CPU memory if VRAM is exhausted.  This is only useful for small amounts of memory as there is a [performance penalty](https://github.com/ollama/ollama/issues/7584#issuecomment-2466715900).  However, in the case where the goal is to reduce OOMs, the amount offloaded will be small and the impact minimal. 
+5. Set [`OLLAMA_NUM_PARALLEL`](https://github.com/ollama/ollama/blob/a4f69a0191b304c204ef074ccd6523f121bfddfe/envconfig/config.go#L249) to 1.
+6. Using a smaller context buffer by reducing [`num_ctx`](https://github.com/ollama/ollama/blob/a4f69a0191b304c204ef074ccd6523f121bfddfe/docs/modelfile.md#valid-parameters-and-values:~:text=mirostat_tau%205.0-,num_ctx,-Sets%20the%20size).
+
+## How can I set the size of the context buffer?
+
+The size can be set in an API call (`"options":{"num_ctx":xxx}}`), in the ollama CLI (`/set parameter num_ctx xxx`), or configured in the model.  When using the OpenAI API compatability endpoint, setting the context size is not currently supported so it must be configured in the model.
+
+There are two ways to configure the context size in a model.  First, use the `/save` command in the ollama CLI:
+```console
+ollama run model-with-normal-context
+>>> /set parameter num_ctx 4096
+>>> /save model-with-4096-context
+>>> /bye
+```
+Second, create a new Modelfile:
+```console
+ollama show --modelfile model-with-normal-context > Modelfile
+echo PARAMETER num_ctx 4096 > Modelfile
+ollama create model-with-4096-context
+```
+After the model has been created, use it in the normal way:
+```console
+curl localhost:11434/api/generate -d "{\"model\":\"model-with-4096-context\",\"prompt\":\"why is the sky blue?\"}"
+```
+```console
+ollama run model-with-4096-context
+```
+If the model is large, the second approach above can cause a lot of disk activity as ollama loads the GGUF file.  This can be prevented by editing the Modelfile, uncommenting the first `FROM` statement and commenting the second `FROM` statement.
```

Comment:
I'm not sure I follow this part, this is suggesting not using a new file but a model which Ollama has already downloaded?